File: Server/src/main/content/global/handlers/item/plugins/WaterSkinPlugin.java
package content.global.handlers.item.plugins;

import core.game.interaction.NodeUsageEvent;
import core.game.interaction.UseWithHandler;
import core.game.node.item.Item;
import core.plugin.Initializable;
import core.plugin.Plugin;

/**
 * The Water skin plugin.
 */
@Initializable
public final class WaterSkinPlugin extends UseWithHandler {

    private final Item FULL_SKIN = new Item(1823);

    private final int[][] data = new int[][]{{1937, 1935}, {1929, 1925}, {1921, 1923}, {227, 229}};

    /**
     * Instantiates a new Water skin plugin.
     */
    public WaterSkinPlugin() {
        super(1825, 1827, 1829, 1831);
    }

    @Override
    public Plugin<Object> newInstance(Object arg) throws Throwable {
        for (int i = 0; i < data.length; i++) {
            addHandler(data[i][0], ITEM_TYPE, this);
        }
        return this;
    }

    @Override
    public boolean handle(NodeUsageEvent event) {
        final Item waterSkin = event.getUsedItem().getName().contains("skin") ? event.getUsedItem() : event.getBaseItem();
        final Item vessil = !event.getUsedItem().getName().contains("skin") ? event.getUsedItem() : event.getBaseItem();
        if (event.getPlayer().getInventory().remove(waterSkin)) {
            event.getPlayer().getInventory().add(vessil.getId() == 227 ? new Item(waterSkin.getId() - 2) : FULL_SKIN);
            for (int i = 0; i < data.length; i++) {
                if (data[i][0] == vessil.getId() && event.getPlayer().getInventory().remove(vessil)) {
                    event.getPlayer().getInventory().add(new Item(data[i][1]));
                }
            }
        }
        return true;
    }

}

File: Server/src/main/content/global/plugins/item/WaterSkinPlugin.kt
package content.global.plugins.item

import core.game.interaction.NodeUsageEvent
import core.game.interaction.UseWithHandler
import core.game.node.item.Item
import core.plugin.Initializable
import core.plugin.Plugin

@Initializable
class WaterSkinPlugin: UseWithHandler(1825, 1827, 1829, 1831) {
    private val FULL_SKIN = Item(1823)

    private val data = arrayOf(intArrayOf(1937, 1935), intArrayOf(1929, 1925), intArrayOf(1921, 1923), intArrayOf(227, 229))

    override fun newInstance(arg: Any): Plugin<Any?> {
        for (i in data.indices) {
            addHandler(data[i][0], ITEM_TYPE, this)
        }
        return this
    }

    override fun handle(event: NodeUsageEvent): Boolean {
        val waterSkin = if (event.usedItem.name.contains("skin")) event.usedItem else event.baseItem
        val vessil = if (!event.usedItem.name.contains("skin")) event.usedItem else event.baseItem
        if (event.player.inventory.remove(waterSkin)) {
            event.player.inventory.add(if (vessil.id == 227) Item(waterSkin.id - 2) else FULL_SKIN)
            for (i in data.indices) {
                if (data[i][0] == vessil.id && event.player.inventory.remove(vessil)) {
                    event.player.inventory.add(Item(data[i][1]))
                }
            }
        }
        return true
    }
}
